### ISSUE TITLE
GRUB not getting colored correctly

### ISSUE THREAD

**Issue by Anomalocaridid** [danth/stylix#83]
For some reason, stylix does not correctly color GRUB.

Rather than using [my color scheme](https://github.com/Anomalocaridid/dotfiles/blob/nixos/stylix.nix), GRUB looks like this:
![Screenshot_nixos-unstable_2023-04-09_14:48:05](https://user-images.githubusercontent.com/29845794/230791233-1bf6249d-7900-4b2f-b24b-fc69b37d9140.png)

I was able to take a screenshot of GRUB because I installed NixOS to a VM.

So far, I have not had any issues theming other programs.

As far as I can tell, this might be related to #61 if the underlying issue's something like an invalid theme causing GRUB to fall back to defaults.

**Comment by danth** [comment]
I wonder if this could be caused by the way you have GRUB installed.

https://github.com/Anomalocaridid/dotfiles/blob/4bafe5cb6ea28385a529997c97a37d173aba8c24/configuration.nix#L34

Perhaps this line?

The GRUB theme works as expected through `nixos-rebuild build-vm` with a basic configuration.

**Comment by Anomalocaridid** [comment]
According to the [NixOS manual (section 2.3.3, step 4)](https://nixos.org/manual/nixos/stable/#sec-installation-manual-installing), setting `boot.loader.grub.device` to `nodev` is necessary for UEFI systems.

Also, just in case it is important, my VM is set up to use UEFI because I eventually plan to use the NixOS config I am setting up on a PC that uses UEFI.

**Comment by Anomalocaridid** [comment]
It works. I think the fact that I was using it in a VM may have been the cause because it works fine on a normal installation. Although it could also be that something changed in NixOS or Stylix since I made this issue.

Sorry for taking so long to get back about this, but shortly after I reported this issue, I ran into some unrelated limitations of the Base16 theming system that irritated me enough to make me stop using Stylix for a while. I didn't get around to giving Stylix another shot until today.
